### ISSUE TITLE
Make lexer (and parser maybe) a separate project?

### ISSUE THREAD

**Issue by wysohn** [TriggerReactor/lsp#1]
I think LSP is a neat idea and has lots of potential.

In addition, it would be awesome if we can separate the project to one that is dedicated only for lexer (and parser if it's needed for LSP) so that we can generate native codes for them.

We can probably use the native codes in the main project as well to speed up lexical analysis/parsing time, and this also will benefit us by having a single point of concern to maintain

**Comment by Sayakie** [comment]
That is how I see it, however, since this project and implementation of TriggerReactor syntax parser that is written with Rust(or TypeScript) will take a long-term time.

I assume that would be good to move the lexer implementation to a different repository first.

**Related to:**
*  https://github.com/TriggerReactor/TriggerReactor/pull/525#issuecomment-1280705893
*  TriggerReactor/TriggerReactor#529

**Comment by wysohn** [comment]
@Sayakie 
Btw, I would highly likely want to participate in this project if you don't mind

It is definitely an awesome opportunity for me to learn a new tech stack (Rust)

And making it usable in both project will be an another challenge haha

Though, 'flying dutchman need a captain,' and I think you'd be a great leader on this project. Let me know if you have some project plan

**Comment by Sayakie** [comment]
Yeah, I'd be happy than you hear from. Feel free to contribute lsp and child repositories.

I'm also learning Rust so it's great for me to co-work this!

No plans in progress yet, but I will have to do on some testable implementation would be released.

**Comment by wysohn** [comment]
> Yeah, I'd be happy than you hear from. Feel free to contribute lsp and child repositories.
> 
> I'm also learning Rust so it's great for me to co-work this!
> 
> No plans in progress yet, but I will have to do on some testable implementation would be released.

Sounds good! Though, we will really need some kind of project management to not step on each other's work. Something similar to the project board in TriggerReactor, but not necessarily has to be that.

I'd expect you to take on the lead?

This one might be one solution: https://github.com/orgs/TriggerReactor/projects/2

Never tried it yet seems to be pretty solid

**Comment by Sayakie** [comment]
LGTM! Or we can make a big project board which should integrate all of side-projects. Visit [Sapphire Project Board](https://github.com/orgs/sapphiredev/projects/1/views/6) for example.

**Comment by Sayakie** [comment]
@wysohn, I added some views that are awesome tables to track our backlog, opened issues, and pull requests for us on [our project board](https://github.com/orgs/TriggerReactor/projects/2).

We have to migrate legacy projects (like v3.4.3) to milestones that fit our project board.

**Comment by wysohn** [comment]
> @wysohn, I added some views that are awesome tables to track our backlog, opened issues, and pull requests for us on [our project board](https://github.com/orgs/TriggerReactor/projects/2).
> 
> We have to migrate legacy projects (like v3.4.3) to milestones that fit our project board.

Oh this is really awesome

Do you mind having some basic setups, so I can follow?

**Comment by Sayakie** [comment]
> Do you mind having some basic setups, so I can follow?

Yeah I'll do whenever have enough time.
